noxfile.py

```diff
@@ -0,0 +1,279 @@
+    )
+
+    session.run("bash", "-c", "cd gcovr/tests && make --silent clean", external=True)
+    args = ["-m", "pytest"]
+    args += coverage_args
+    args += session.posargs
+    if "--" not in args:
+        args += ["--", "gcovr", "doc/examples"]
+    session.run("python", *args)
+
+
+@nox.session
+def build_wheel(session: "nox.session") -> None:
+    session.install("wheel", "twine")
+    session.run("python", "setup.py", "sdist", "bdist_wheel")
+    session.run("twine", "check", "dist/*", external=True)
+
+
+@nox.session
+def upload_wheel(session: "nox.session") -> None:
+    session.install("twine")
+    session.run("twine", "upload", "dist/*", external=True)
+
+
+def docker_container_id(version: str) -> None:
+    return "gcovr-qa-{}".format(version)
+
+
+@nox.session
+def docker_qa_build(session: "nox.session") -> None:
+    session_id = "docker_qa__build_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def docker_qa_build_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_build_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def docker_qa_build_version(session: "nox.session", version: str) -> None:
+    set_environment(session, version, False)
+    session.run(
+        "bash",
+        "-c",
+        " ".join(
+            [
+                "docker",
+                "build",
+                "--tag",
+                docker_container_id(version),
+                "--build-arg",
+                "USERID={}".format(os.geteuid()),
+                "--build-arg",
+                "CC=${CC}",
+                "--build-arg",
+                "CXX=${CXX}",
+                "--file",
+                "admin/Dockerfile.qa",
+                ".",
+            ]
+        ),
+        external=True,
+    )
+
+
+@nox.session
+def docker_qa_run(session: "nox.session") -> None:
+    session_id = "docker_qa_run_version({})".format(GCC_VERSION2USE)
+    session.log("Notify session {}".format(session_id))
+    session.notify(session_id)
+
+
+@nox.session
+def docker_qa_run_all_versions(session: "nox.session") -> None:
+    for version in GCC_VERSIONS:
+        session_id = "docker_qa_run_version({})".format(version)
+        session.log("Notify session {}".format(session_id))
+        session.notify(session_id)
+
+
+@nox.session
+@nox.parametrize("version", [nox.param(v, id=v) for v in GCC_VERSIONS])
+def docker_qa_run_version(session: "nox.session", version: str) -> None:
+    set_environment(session, version, False)
+    session.run(
+        "bash",
+        "-c",
+        " ".join(
+            [
+                "docker",
+                "run",
+                "--rm",
+                "-e",
+                "TESTOPTS",
```

Comment:
No TESTOPTS variable exists. Instead, we should probably stick posargs into an environment variable, pass it to the container like this, and then manually pass it to Nox' posargs in the Dockerfile.